### PR TITLE
Feature/dead system

### PR DESCRIPTION
**Pull request by baptiste-arnoulet**
Create a checkPoint and respawn system.
 
 - Add CheckPoint Component
 - CheckPoint have a render (cute star ;) )

Add Trigger collision system for checkPoint

### CODE REVIEW

File: Projet/gameplay/CheckPoint.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+#include "core/Component.h"
+#include "physic/CapsuleCollider.h"
+#include "physic/RigidBody.h"
+#include "physic/SphereCollider.h"
+
+class CheckPoint : public Pitbull::Component
+{
+public:
+	CheckPoint(Pitbull::Actor* Parent);
```
Comment (carobouette) [inline Projet/gameplay/CheckPoint.h:15]: it will be better with one or two comments

File: Projet/physic/ContactHandler.h [hunk 2]
```diff
@@ -14,12 +14,13 @@ class ContactHandler : public physx::PxSimulationEventCallback, public physx::Px
     virtual void onConstraintBreak(physx::PxConstraintInfo* constraints, physx::PxU32 count) {}
```
Comment (carobouette) [inline Projet/physic/ContactHandler.h:14]: MaJuScUlEs

File: Projet/render/GameFactory.h [hunk 3]
```diff
@@ -65,4 +65,10 @@ class GameFactory : public Singleton<GameFactory>
 	///	<param name="OuterRadius"> : outerRadius of the light </param>
 	void CreateLights(DirectX::XMFLOAT3 Pos, DirectX::XMFLOAT3 Specular, DirectX::XMFLOAT3 Roughness, float Intensity = 0.f, float InnerRadius = 0.f, float OuterRadius = 0.f);
 
+	/// <summary>
+	/// Create an enemy
+	/// </summary>
+	///	<param name="Pos"> : position of the enemy  </param>
+	///	<param name="Scale"> : scale of the enemy  </param>
```
Comment (carobouette) [inline Projet/render/GameFactory.h]: comment dont fit with code

File: Projet/physic/ContactHandler.h (hunk 2 above)
Comment (baptiste-arnoulet) [reply Projet/physic/ContactHandler.h:14]: I not touch this, but I think we want to keep a same format as the onContact function, and they don't take uppercase.

Comment (baptistegasser) [reply Projet/physic/ContactHandler.h:14]: +1 we should keep physx convention when interfacing with it like here

**Review by carobouette — APPROVED** [review]
youpii

**Pull request merged**